docs: clean up README.md and status.md (#99)
Stedi/jsonata-rs: docs/status.md

# Status

There are several issues to be resolved:

> - We've tried to implement structural sharing of the input and output values with minimal heap allocations. This was a lot of effort working out the lifetimes, but we are not sure it was worth it. We may consider deleting Bumpalo in the future.

Reviewer: ```suggestion
- This implementation attempts structural sharing of the input and output values with minimal heap allocations. This was a lot of effort working out the lifetimes, but we are not sure it was worth it. We may consider deleting Bumpalo in the future.
```

- The code is too spaghetti in some places and could be more Rust-idiomatic.
- We have made a couple of optimization passes, but there are still lots of opportunities for optimization.

> - There are obviously still a bunch of missing features - We are aiming for feature parity with the reference implementation (within reason)

Reviewer: ```suggestion
- There are obviously still a bunch of missing features. We're aiming for feature parity with the reference implementation wherever feasible.
```
